ios/RNSContentScrollViewDetector.mm

```diff
@@ -0,0 +1,101 @@
+#import "RNSContentScrollViewDetector.h"
+#import <React/RCTLog.h>
+#import <react/renderer/components/rnscreens/ComponentDescriptors.h>
+
+namespace react = facebook::react;
+
+@implementation RNSContentScrollViewDetector {
+  UIScrollView *_scrollView;
+  UIView *_linkedView;
+}
+
+- (void)didMoveToWindow
+{
+  if (self.window == nil) {
+    [self unregisterContentScrollViewInHierarchy];
+  } else {
+    if ((_scrollView = [self findContentScrollView])) {
+      [self registerContentScrollViewInHierarchy];
+    }
+  }
+}
+
+- (UIScrollView *)findContentScrollView
+{
+  auto scrollView = [RNSScrollViewFinder findScrollViewInFirstDescendantChainFrom:self];
+  if (scrollView) {
+    RCTLogInfo(@"Found content ScrollView");
+    return scrollView;
+  }
+
+  return nullptr;
+}
+
+- (void)registerContentScrollViewInHierarchy
+{
+  if (_linkedView != nullptr) {
+    RCTLogWarn(
+        @"Content ScrollView has already been registered. Make sure to only have one ScrollViewWrapper for content ScrollView.");
+    [self unregisterContentScrollViewInHierarchy];
+  }
+
+  UIView *parent = self.superview;
+  while (parent) {
+    if ([parent respondsToSelector:@selector(findContentScrollView)]) {
```

Comment:
what if there will be another library that will use this (imo) common name for selector?